File: server/src/main/java/tech/wajs/battleship/enums/LocationStatus.java
/*
 * Copyright (c) 2022 Konrad Wajs, All rights reserved.
 */

package tech.wajs.battleship.enums;

public enum LocationStatus {
    WRECK("W"),
    OCCUPIED("O"),
    MISS("M"),
    EMPTY(" ");

    public String abbrev;

    LocationStatus(String abbrev) {
        this.abbrev = abbrev;
    }
}

File: client/battleship/src/shared/enums/LocationStatus.ts
export enum LocationStatus {
    WRECK = "WRECK",
    MISS = "MISS",
    EMPTY = "EMPTY"
}

export interface LocationType {
    abbrev: string
}

export const LocationStatusProperties = new Map<LocationStatus, LocationType>([
    [LocationStatus.WRECK, {abbrev: "O"}],
    [LocationStatus.MISS, {abbrev: "X"}],
    [LocationStatus.EMPTY, {abbrev: ""}],
])

export const getLocationStatusName = (locationStatus: LocationStatus): string | undefined =>
    LocationStatusProperties.get(locationStatus)?.abbrev

